openpype/hosts/blender/api/capture.py

```diff
@@ -149,12 +129,29 @@ def isolate_objects(window, objects, focus=None):
     deselect_all()
 
 
-def _apply_options(entity, options):
-    for option, value in options.items():
-        if isinstance(value, dict):
-            _apply_options(getattr(entity, option), value)
+def _apply_settings(entity, settings):
+    for option, value in settings.items():
+        if hasattr(entity, option):
+            if isinstance(value, dict):
+                _apply_settings(getattr(entity, option), value)
+            else:
+                setattr(entity, option, value)
+
+
+def _get_current_settings(entity, settings):
+    current_settings = {}
+    for option in settings.copy():
```

Comment:
I don't get why you copy it, it is pretty slow to run it at every loop. If this is to avoid index jump, you could use `reverse(settings)`

Reply:
👍 I no longer need copy and the pop options with hasattr in the apply_settings function.